pkg/kubernetes/client/v1alpha1/client.go

```diff
@@ -91,12 +91,25 @@ type getKubernetesClientsetFunc func(*rest.Config) (*kubernetes.Clientset, error
 
 // Client provides common kuberenetes client operations
 type Client struct {
-	IsInCluster            bool                       // flag to let client point to its own cluster
-	getInClusterConfig     getInClusterConfigFunc     // handle to get in cluster config
-	buildConfigFromFlags   buildConfigFromFlagsFunc   // handle to get desired kubernetes config
-	getKubernetesClientset getKubernetesClientsetFunc // handle to get kubernetes clienset
-	getKubeMasterIP        getKubeMasterIPFunc        // handle to get kubernetes master IP
-	getKubeConfigPath      getKubeConfigPathFunc      // handle to get kubernetes config path
+	IsInCluster    bool   // flag to let client point to its own cluster
+	KubeConfigPath string // kubeconfig path to get kubernetes clientset
+
+	// Below functions are useful during mock
+
+	// handle to get in cluster config
+	getInClusterConfig getInClusterConfigFunc
+
+	// handle to get desired kubernetes config
+	buildConfigFromFlags buildConfigFromFlagsFunc
+
+	// handle to get kubernetes clienset
+	getKubernetesClientset getKubernetesClientsetFunc
+
+	// handle to get kubernetes master IP
+	getKubeMasterIP getKubeMasterIPFunc
+
+	// handle to get kubernetes config path
+	getKubeConfigPath getKubeConfigPathFunc
```

Comment:
What's the need for this property

Reply:
@AmitKumarDas we are getting the kubeconfigpath from ENV using above typed function